src/app/state/types.ts

```diff
@@ -49,3 +49,37 @@ export type CurrentView =
   | { type: 'story-options'; subview: 'story-metadata' | 'story-wide-code' | 'asset-upload' }
   | { type: 'scene'; id: string; subview: 'scene-options' | '3d-view' }
   | { type: 'preview'; previous?: CurrentView };
+
+export type Visibility = 'public' | 'private';
+
+export interface Creator {
+  email: string;
+  id: string;
```

Comment:
I wonder if all users should get their own folder: `{user-id}/sessions|states/{object-id}.{object-extension}`?

Reply:
Making an object from private to public changes its location, meaning the link would change. Do we want this?

Also, this will make it quite slow to list all public states across users -- I think we will skip this feature (listing "recent public states" on the home page) for now .... looks like we would need a SQL DB over the artifacts and that would be out of scope

Reply:
Let's focus on just the sharing aspects for now without having the ability to search. To do that properly I don't think a a separate json index would work (it would break with too many entries, concurrency, etc).